src/test/ui/parser/issues/issue-94340.stderr

```diff
@@ -0,0 +1,28 @@
+error: an inner attribute is not permitted in this context
+  --> $DIR/auxiliary/issue-94340-inc.rs:2:1
+   |
+LL | #![deny(rust_2018_idioms)]
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: inner attributes, like `#![no_std]`, annotate the item enclosing them, and are usually found at the beginning of source files
```

Comment:
In the linked issue I authored, I criticize the part _are usually found at the beginning of source files_. On one hand, I'd like to add sth. akin to _except inside `include!`'d files_ to resolve potential confusion. On the other hand, this would make this rather general note too verbose and specific.

Reply:
The problem is that macros can't expand to outer attributes (not only `include!`, but *any* macros). We can extend the wording to mention that, or (ideally), we would check that we're in macro context and be explicit about it (saying "inner attributes are not permitted in macro context", similar in the label, the note can remain the same).

@petrochenkov, what would be the correct way of checking whether one of these comes from a macro expansion? Just checking their span's macro backtrace isn't empty would do the trick and be 100% accurate, right?

Reply:
Oh, if that works, that would be great. I like the wording _macro context_.

Reply:
Ok, I tried using [`Span::from_expansion`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.Span.html#method.from_expansion) to detect if the inner attributes are macro-generated. However, the method returned `false`. I think this is because [`expand_include`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_builtin_macros/source_util/fn.expand_include.html) just uses the parser method `parse_item` not identifying itself as a macro. Not sure if that's how it works. Maybe the reason is the way the SourceFile is loaded: Maybe it's just included as a regular source file and not marked as a “special” file (because it's not special?).

`attr_sp.from_expansion()` returns `false`:
https://github.com/rust-lang/rust/blob/b6ab1fae73a14db17a59e81c532a2b8f048ac159/compiler/rustc_parse/src/parser/attr.rs#L217

Potentially, `from_expansion` is not what I want. I should probably look into [`Span::ctxt`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.Span.html#method.ctxt) and see if there's anything interesting (which I doubt since it has to equal `SyntaxContext::root()` according to `from_expansion`.

Reply:
> Maybe the reason is the way the include!d [SourceFile](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/struct.SourceFile.html) is loaded: Maybe it's just included as a regular source file and not marked as a “special” file (because it's not special?).

I would consider that would be a bug, but currently harmless, because nothing *cares* about the included code being included, I think.